js/ui/appDisplay.js

```diff
@@ -2198,7 +2198,8 @@ var AppFolderDialog = GObject.registerClass({
             coordinate: Clutter.BindCoordinate.SIZE,
         }));
 
-        this._folder.connect('changed::name', () => this._syncFolderName());
+        this._folderNameChangedId = this._folder.connect(
+            'changed::name', () => this._syncFolderName());
```

Comment:
When does the folder dialog outlives the folder settings? All their lifetimes seem tightly tied

Reply:
As above we never destory the folder explicitly when the FolderView/Dialog/Icon is destroyed and rely on GC.

Reply:
The dialog is explicitly destroyed together with the icon: https://github.com/endlessm/gnome-shell/blob/master/js/ui/appDisplay.js#L1895

Reply:
Yes, the dialog is but not this._folder (the gsettings).

Reply:
Kept this one but can drop also if you prefer.

Reply:
Fwiw, upstream/FolderIcon already connects to folder changed signal and disconnects when FolderIcon is destroyed, but not for the changed:name.

Reply:
https://github.com/endlessm/gnome-shell/blob/master/js/ui/appDisplay.js#L1887-L1888